### PR TITLE
Run

### PR DESCRIPTION
**Pull request by nicolasaunai**
#464 

On top of #490  and #478

### CODE REVIEW

**Comment by PhilipDeegan** [comment]
LGTM is giving a different reason for the "x != x" bit now + one unused var

File: pyphare/pyphare/pharesee/hierarchy.py [hunk 1]
```diff
@@ -409,6 +409,25 @@ def levelNbr(self, time):
...
+            for ilvl, lvl in levels.items():
+                for patch in lvl.patches:
+                    it_is &= qties == list(patch.patch_datas.keys())
+        return it_is
+
+    def _quantities(self):
+        return list(self.level(0).patches[0].patch_datas.keys())
+
+    def quantities(self):
+        assert self.is_homogeneous()
```
Comment (PhilipDeegan) [inline pyphare/pyphare/pharesee/hierarchy.py:429]: this doesn't exist for "python -O" executions

maybe not an issue

Comment (nicolasaunai) [reply pyphare/pyphare/pharesee/hierarchy.py:429]: I don't understand what you mean

Comment (PhilipDeegan) [reply pyphare/pyphare/pharesee/hierarchy.py:429]: should it be a "raise RuntimeError" or an assert. as the assert is disabled when running with "python3 -O script.py"

**Review by PhilipDeegan — APPROVED** [review]

File: pyphare/pyphare/pharesee/hierarchy.py (hunk 1 above)
Comment (nicolasaunai) [reply pyphare/pyphare/pharesee/hierarchy.py:429]: ok I didn't know that. I'll change it to a raise RunTimeError then

**Pull request merged**